src/app/shared-services/staypoint/staypoint.service.ts

```diff
@@ -0,0 +1,223 @@
+import { Injectable } from '@angular/core'
+import { SqliteService } from '../../shared-services/db/sqlite.service'
+import haversine from 'haversine-distance'
+import {
+  Trajectory,
+  TrajectoryData,
+  TrajectoryType,
+} from 'src/app/model/trajectory'
+import { StayPointData, StayPoints } from 'src/app/model/staypoints'
+import { TrajectoryService } from '../trajectory/trajectory.service'
+import { take } from 'rxjs/operators'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StaypointService {
+  // for meaning of these two parameters, please see detectStayPoints() documentation
+  // if you change one or both, please also update the associated detected staypoints in staypoint.service.spec.ts
+  readonly DIST_THRESH_METERS = 100
+  readonly TIME_THRESH_MINUTES = 15
```

Comment:
You could make these values static. That way the could be accessed from outside if needed (e.g. in the `spec.ts`)